src/adr/0030-separate-attachment-content-and-metadata.md

```diff
@@ -0,0 +1,84 @@
+ - **Message** exposes its list of MessageAttachment when it is read with FetchType Full..
+ - **Blob** represents some downloadable content, and can be either an attachment or a message. Blob has a byte array 
+ payload too.
+ 
+The following classes works with the aforementioned POJOs:
+ 
+ - **AttachmentMapper** and **AttachmentManager** are responsible of storing and retrieving an attachment content.
+ - **BlobManager** is used by JMAP to allow blob downloads.
+ - Mailbox search exposes attachment content related criteria. These criteria are used by the JMAP protocol.
+
+This organisation causes attachment content to be loaded every time a message is fully read (which happens for instance
+when you open a message using JMAP) despite the fact that it's not needed.
+
+Also, the content being loaded "at once", we allocate memory space to store the whole attachment, which is sub-optimal.
+
+To be noted that JPA and maildir mailbox implementations do not support attachment storage. To retrieve attachments of a 
+message, these implementations parse the message to extract their attachments.
+
+Cassandra mailbox prior schema version 4 stores attachment and its metadata in the same table, but from version 5 relies 
+on the blobStore to store the attachment content.
+
+## Decision
+
+Enforce cassandra schema version to be 5 from James release 3.5.0. This allows to drop attachment management prior version
+5.
+
+We will re-organize the attachment POJOs: 
+
+ - **Attachment** should hold an attachmentId, a content type, and a size. It will no longer hold the content.
+ - **MessageAttachment** : composes an attachment with it's disposition within a message (cid, inline and name)
+ - **Blob** would no longer hold the content as a byte array but rather a content retriever (`Supplier<InputStream>`)
+ - **ParsedAttachment** is the direct result of attachment parsing, and composes a **MessageAttachment** and the 
+ corresponding content as byte array. This class is only relied upon when saving a message in mailbox. This is used as 
+ an output of `MessageParser`.
+
+Some adjustments are needed on class working with attachment:
+
+ - **AttachmentMapper** and **AttachmentManager** needs to allow from an attachmentId to retrieve the attachment content
+ as an `InputStream`. This is done through a separate `AttachmentLoader` interface.
+ - **AttachmentMapper** and **AttachmentManager** needs the Attachment and its content to persist an attachment
+ - **MessageManager** then needs to return attachment metadata as a result of Append operation.
+ - **InMemoryAttachmentMapper** needs to store attachment content separately.
+ - **MessageStorer** will take care of storing a message on the behalf of `MessageManager`. This enables to determine if 
+ attachment should be parsed or not on an implementation aware fashion, saving attachment parsing upon writes for JPA 
+ and Maildir.
+ 
+Maildir and JPA no longer support attachment content loading.
```

Comment:
why?

Reply:
Because it's used nowhere cf JMAP only requires attachment content loading.

This could be developped with coupling on the attachment naming strategy.